src/main/java/com/avispl/symphony/dal/avdevices/camera/panasonic/awue150/CameraPanasonicAWUE150Communicator.java

```diff
@@ -0,0 +1,1779 @@
+				currentTiltCommandValue = DeviceConstant.MIN_TILT_DOWN_API_VALUE + (int) cachedPanTiltControlSpeed;
+
+				if (currentPanCommandValue < DeviceConstant.MIN_PAN_RIGHT_API_VALUE) {
+					currentPanCommandValue = DeviceConstant.MIN_PAN_RIGHT_API_VALUE;
+				}
+				if (currentTiltCommandValue > DeviceConstant.MAX_PAN_LEFT_API_VALUE) {
+					currentTiltCommandValue = DeviceConstant.MAX_PAN_LEFT_API_VALUE;
+				}
+				command = Command.PAN_TILT
+						.concat(String.valueOf(currentPanCommandValue))
+						.concat(String.format(DeviceConstant.TWO_NUMBER_FORMAT, currentTiltCommandValue));
+//				performCameraControl(Command.PAN_TILT_DOWN_RIGHT_DEFAULT, controllableProperty, DeviceURL.CAMERA_PTZ_CONTROL, true);
+				performCameraControl(command, controllableProperty, value, DeviceURL.CAMERA_PTZ_CONTROL, true);
+				performCameraControl(Command.PAN_TILT_STOP, controllableProperty, value, DeviceURL.CAMERA_PTZ_CONTROL, true);
+
+				// camera need 2s to apply new value
+				try {
+					Thread.sleep(2000);
+				} catch (InterruptedException e) {
+					e.printStackTrace();
+				}
+				retrieveLiveCameraInfo(stats);
+				populatePanTiltControls(stats, advancedControllableProperties);
+				break;
+			case DOWN_LEFT:
+				currentCommandValue = DeviceConstant.MAX_PAN_LEFT_API_VALUE - (int) cachedPanTiltControlSpeed;
+				if (currentCommandValue < DeviceConstant.MIN_PAN_LEFT_API_VALUE) {
+					currentCommandValue = DeviceConstant.MIN_PAN_LEFT_API_VALUE;
+				}
+				command = Command.PAN_TILT
+						.concat(String.format(DeviceConstant.TWO_NUMBER_FORMAT, currentCommandValue))
+						.concat(String.format(DeviceConstant.TWO_NUMBER_FORMAT, currentCommandValue));
+//				performCameraControl(Command.PAN_TILT_DOWN_LEFT_DEFAULT, controllableProperty, DeviceURL.CAMERA_PTZ_CONTROL, true);
+				performCameraControl(command, controllableProperty, value, DeviceURL.CAMERA_PTZ_CONTROL, true);
+				performCameraControl(Command.PAN_TILT_STOP, controllableProperty, value, DeviceURL.CAMERA_PTZ_CONTROL, true);
+
+				// camera need 2s to apply new value
+				try {
+					Thread.sleep(2000);
+				} catch (InterruptedException e) {
+					e.printStackTrace();
+				}
+				retrieveLiveCameraInfo(stats);
+				populatePanTiltControls(stats, advancedControllableProperties);
+				break;
+			case HOME:
+				performCameraControl(Command.HASH.concat(Command.HOME_PRESET), value, controllableProperty, DeviceURL.CAMERA_PTZ_CONTROL, true);
+				break;
+			case PT_SPEED:
+				cachedPanTiltControlSpeed = Float.parseFloat(value);
+				populatePanTiltControls(stats, advancedControllableProperties);
+				break;
+			default:
+				throw new IllegalStateException(String.format("preset control %s is not supported.", controllableProperty));
+		}
+		populatePresetControls(stats, advancedControllableProperties);
+	}
+
+	/**
+	 * This method is used to populate pan tilt control
+	 */
+	private String convertPanTiltPositionFromApiToUiValue(String panTiltData) {
```

Comment:
@param and @return are missing in javadoc

Reply:
I fixed it